pkg/controller/networkconfig/networkconfig_controller.go

```diff
@@ -185,15 +189,24 @@ func (r *ReconcileNetworkConfig) Reconcile(request reconcile.Request) (reconcile
 	}
 	objs = append([]*uns.Unstructured{app}, objs...)
 
-	// Set up the DaemonSet reconciler before we start creating the DaemonSets
+	// Set up the Pod reconciler before we start creating DaemonSets/Deployments
 	r.status.SetConfigSuccess()
 	daemonSets := []types.NamespacedName{}
+	deployments := []types.NamespacedName{}
 	for _, obj := range objs {
 		if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "DaemonSet" {
 			daemonSets = append(daemonSets, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+		} else if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "Deployment" {
+			deployments = append(deployments, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
 		}
 	}
-	r.daemonSetReconciler.SetDaemonSets(daemonSets)
+	r.status.SetDaemonSets(daemonSets)
+	r.status.SetDeployments(deployments)
+
+	allResources := []types.NamespacedName{}
```

Comment:
I realize this is a bit pathological, but you could totally have a NamespacedName collision between daemonsets and deployments.

Reply:
Especially if you moved from one Kind to the other.

Reply:
Ahh, now I see why that doesn't really matter.

Reply:
I started working on a controller-runtime PR to make it include a `GroupVersionKind` in `reconcile.Request`. But as you figured out, it doesn't actually matter here.